ui/src/utils/clipboard_data.rs

```diff
@@ -0,0 +1,169 @@
+#[cfg(target_os = "macos")]
+use super::macos_clipboard::MacOSClipboard;
+#[cfg(target_os = "linux")]
+use super::verify_valid_paths::decoded_pathbufs;
+use arboard::Clipboard as Arboard;
+#[cfg(target_os = "windows")]
+use clipboard_win::{formats, get_clipboard, set_clipboard};
+use image::DynamicImage;
+use image::ImageBuffer;
+use image::ImageOutputFormat;
+use image::RgbaImage;
+use std::error::Error;
+use std::io::BufWriter;
+#[cfg(target_os = "linux")]
+use std::path::Path;
+use std::path::PathBuf;
+use tempfile::TempDir;
+
+pub enum ClipboardDataType {
+    File,
+    String,
+}
+
+/// It will verify if data in clipboard are local paths of files to upload them.
+///
+/// if not, it will grab pixels of image data in clipboard and transform them into
+/// an image file to be possible to upload.
+pub fn get_files_path_from_clipboard() -> Result<Vec<PathBuf>, Box<dyn std::error::Error>> {
+    #[cfg(target_os = "windows")]
+    {
+        let file_path: Vec<PathBuf> = get_clipboard(formats::FileList {})
+            .unwrap_or_default()
+            .into_iter()
+            .map(PathBuf::from)
+            .collect();
+        println!("file_path: {:?}", file_path);
+        if !file_path.is_empty() {
+            return Ok(file_path);
+        }
+    }
+    #[cfg(target_os = "macos")]
+    {
+        let macos_clipboard = MacOSClipboard::new()?;
+        let file_path = macos_clipboard
+            .get_files_path_from_clipboard()
+            .unwrap_or_default();
+        if !file_path.is_empty() {
+            return Ok(file_path);
+        }
+    }
+
+    #[cfg(target_os = "linux")]
+    {
+        if let Ok(mut clipboard) = Arboard::new() {
+            let clipboard_text = clipboard.get_text().unwrap_or_default();
+            let paths_vec: Vec<PathBuf> = clipboard_text.lines().map(PathBuf::from).collect();
+            let is_valid_paths = match paths_vec.first() {
+                Some(first_path) => Path::new(first_path).exists(),
+                None => false,
+            };
+            if is_valid_paths {
+                let files_path = decoded_pathbufs(paths_vec);
+                if !files_path.is_empty() {
+                    return Ok(files_path);
+                }
+            }
+        }
+    }
+
+    let image_from_clipboard = check_image_pixels_in_clipboard().unwrap_or(Vec::new());
```

Comment:
wouldnt it be better to reuse the clipboard above instead of making multiple calls?